frontend/src/utils/fetchMajorsAndPlans.ts

```diff
@@ -69,8 +70,10 @@ export function fetchMajorsAndPlans() {
       })
         .then(res => res.json())
         .then(res => {
-          if (res.error) {
-            throw res.error;
+          if (res.statusCode >= 400) {
+            history.replace(history.location.pathname, {
+              errorStatusCode: res.statusCode,
+            });
```

Comment:
This will catch all the errors higher than 400 from the api call and put them into history.